src/server/forum_user_handler.cpp

```diff
@@ -184,95 +139,21 @@ fuh::ban_info fuh::user_is_banned(const std::string& name, const std::string& ad
 	//       prepared SQL statement API right now. However, they are basically
 	//       never used on forums.wesnoth.org, so this shouldn't be a problem
 	//       for the time being.
-	//
-
-	// NOTE: A ban end time of 0 is a permanent ban.
-	const std::string& is_extant_ban_sql =
-		"ban_exclude = 0 AND (ban_end = 0 OR ban_end >=" + std::to_string(std::time(nullptr)) + ")";
-
-	// TODO: retrieve full ban info in a single statement instead of issuing
-	//       separate queries to check for a ban's existence and its duration.
-
-	try {
-		if(!addr.empty() && prepared_statement<bool>("SELECT 1 FROM `" + db_banlist_table_ + "` WHERE UPPER(ban_ip) = UPPER(?) AND " + is_extant_ban_sql, addr)) {
+	ban_check b = conn.get_ban_info(name, addr);
+	switch(b.get_ban_type())
+	{
+		case 1:
 			LOG_UH << "User '" << name << "' ip " << addr << " banned by IP address\n";
-			return retrieve_ban_info(BAN_IP, addr);
-		}
-	} catch(const sql_error& e) {
-		ERR_UH << "Could not check forum bans on address '" << addr << "' :" << e.message << '\n';
-		return {};
+			return { BAN_IP, b.get_ban_duration() };
+		case 2:
+			LOG_UH << "User '" << name << "' uid " << b.get_user_id() << " banned by uid\n";
+			return { BAN_USER, b.get_ban_duration() };
+		case 3:
+			LOG_UH << "User '" << name << "' email " << b.get_email() << " banned by email address\n";
+			return { BAN_EMAIL, b.get_ban_duration() };
+		default:
+			return {};
```

Comment:
This cannot really happen if a ban was found, right? Maybe it should be an assert or at least a log message.

Reply:
This is the default behavior for when a ban isn't found.